server/src/main/java/org/opensearch/identity/IdentityService.java

```diff
@@ -56,4 +64,77 @@ public Subject getSubject() {
     public TokenManager getTokenManager() {
         return identityPlugin.getTokenManager();
     }
+
+    public static IdentityService getInstance() {
+        if (instance == null) {
+            new IdentityService(Settings.EMPTY, List.of());
+        }
+        return instance;
+    }
+
+    /**
+     * Sets the application list known to the IdentityService
+     * @param applicationList A list of all principals for known applications
+     */
+    public void setApplicationList(List<Principal> applicationList) {
```

Comment:
This shouldn't be settable by any other classes, I think we should query the extension manager directly instead

Reply:
I think this needs to be in the IdentityService in order to support multiple types of application instead of just extensions. If we directly put this in the ExtensionManager, we lock ourselves into only having applications which conform to the specifications of extensions. If we include the ApplicationList logic as part of the IdentityService, then we can support various types of applications easier I believe. For example, a plugin or other process which would not fall into the extension category could be registered as an application. 

Perhaps I am misunderstanding things though? What do you think?